File: src/main/java/com/github/svegon/capi/mixininterface/IClientPlayNetworkHandler.java
package com.github.svegon.capi.mixininterface;

import net.minecraft.client.network.ClientDynamicRegistryType;
import net.minecraft.client.network.PlayerListEntry;
import net.minecraft.client.world.ClientWorld;
import net.minecraft.registry.CombinedDynamicRegistries;

import java.util.Map;
import java.util.UUID;

public interface IClientPlayNetworkHandler {
    Map<UUID, PlayerListEntry> playerListEntries();

    void setWorld(ClientWorld world);

    ClientWorld.Properties getWorldProperties();

    void setWorldProperties(ClientWorld.Properties properties);

    void setCombinedDynamicRegistries(CombinedDynamicRegistries<ClientDynamicRegistryType> combinedDynamicRegistries);
}

File: src/client/kotlin/io/github/svegon/mclientapi/client/mixinterface/IClientPlayNetworkHandler.kt
package io.github.svegon.mclientapi.client.mixinterface

import io.github.svegon.mclientapi.mixininterface.IClientPlayPacketListener
import net.minecraft.client.network.PlayerListEntry
import net.minecraft.client.world.ClientWorld
import net.minecraft.registry.DynamicRegistryManager
import java.util.*

interface IClientPlayNetworkHandler : IClientPlayPacketListener {
    val playerListEntries: Map<UUID, PlayerListEntry>

    var world: ClientWorld?

    var worldProperties: ClientWorld.Properties?

    var combinedDynamicRegistries: DynamicRegistryManager.Immutable
}
